### PR TITLE
FAC API course search operation refactor

### PR DESCRIPTION
**Pull request by gunndabad**

### CODE REVIEW

File: src/Dfc.CourseDirectory.Core/Search/CourseSearchQuery.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+        public string OrderBy { get; set; }
+
+        public int Size { get; set; }
+
+        public int? Skip { get; set; }
+
+        public (string SearchText, SearchOptions Options) GenerateSearchQuery()
+        {
+            var filter = string.Join(" and ", (Filters ?? Array.Empty<string>()).Append("Status eq 1"));
```
Comment (jameslawesynergy) [inline src/Dfc.CourseDirectory.Core/Search/CourseSearchQuery.cs:29]: `Enumerable.Empty<string>()`?

Comment (jameslawesynergy) [reply src/Dfc.CourseDirectory.Core/Search/CourseSearchQuery.cs:29]: Use `nameof(Course.Status)`?

File: src/Dfc.CourseDirectory.Core/Search/CourseSearchQuery.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+            var searchOptions = new SearchOptions()
+            {
+                Filter = filter,
+                IncludeTotalCount = true,
+                ScoringProfile = RegionBoostScoringProfileName,
+                SearchFields =
+                {
+                    "QualificationCourseTitle",
```
Comment (jameslawesynergy) [inline src/Dfc.CourseDirectory.Core/Search/CourseSearchQuery.cs:38]: Use `nameof()`

**Review by jameslawesynergy — APPROVED** [review]

File: src/Dfc.CourseDirectory.Core/Search/AzureSearch/ServiceCollectionExtensions.cs [hunk 3]
```diff
@@ -7,13 +7,19 @@ namespace Dfc.CourseDirectory.Core.Search.AzureSearch
...
+            this IServiceCollection services,
+            Uri endpoint,
+            string key,
+            string indexName,
+            Action<SearchClientOptions> configureOptions = null)
         {
-            services.AddSingleton<ISearchClient<TResult>>(s =>
-                new AzureSearchClient<TResult>(
-                    new SearchClient(endpoint, indexName, new AzureKeyCredential(key))));
+            var options = new SearchClientOptions();
```
[the remark this answers is not in the document]
Comment (gunndabad) [reply src/Dfc.CourseDirectory.Core/Search/AzureSearch/ServiceCollectionExtensions.cs:17]: No:
```cs
        public SearchClient(
            Uri endpoint,
            string indexName,
            AzureKeyCredential credential) :
            this(endpoint, indexName, credential, null)
        {
        }
```

[the remark this answers is not in the document]
Comment (gunndabad) [reply src/Dfc.CourseDirectory.Core/Search/AzureSearch/ServiceCollectionExtensions.cs:17]: No:
```cs
        public SearchClient(
            Uri endpoint,
            string indexName,
            AzureKeyCredential credential) :
            this(endpoint, indexName, credential, null)
        {
        }

        public SearchClient(
            Uri endpoint,
            string indexName,
            AzureKeyCredential credential,
            SearchClientOptions options)
        {
            //...
            options ??= new SearchClientOptions();
            //...
        }
```

File: src/Dfc.CourseDirectory.Core/Search/CourseSearchQuery.cs (hunk 1 above)
Comment (gunndabad) [reply src/Dfc.CourseDirectory.Core/Search/CourseSearchQuery.cs:29]: `Array.Empty<T>()` and `Enumerable.Empty<string>()` are the same other than return type: https://github.com/dotnet/runtime/blob/6072e4d3a7a2a1493f514cdf4be75a3d56580e84/src/libraries/System.Linq/src/System/Linq/Enumerable.SizeOpt.cs#L10

Comment (jameslawesynergy) [reply src/Dfc.CourseDirectory.Core/Search/CourseSearchQuery.cs:29]: That doesn't surprise me, but given that you're using it in place of an `IEnumerable<string>` it seems the more appropriate choice.

**Pull request merged**